### PR TITLE
Scalability study ssc

### PR DESCRIPTION
**Pull request by ahellander**
Changes in client-combiner allocation logic as disscussed. Clients get allocated upon connection. 
Many bugfixes in Reducer control.

### CODE REVIEW

File: fedn/cli/run_cmd.py [hunk 1]
```diff
@@ -49,10 +54,11 @@ def reducer_cmd(ctx, discoverhost, discoverport, token, name):
 @click.option('-h', '--hostname', required=True)
 @click.option('-i', '--port', required=True)
 @click.option('-s', '--secure', required=False, default=True)
+@click.option('-c', '--max_clients', required=False,default=8)
 @click.pass_context
-def combiner_cmd(ctx, discoverhost, discoverport, token, name, hostname, port, secure):
+def combiner_cmd(ctx, discoverhost, discoverport, token, name, hostname, port, secure,max_clients):
```
Comment (morganekmefjord) [inline fedn/cli/run_cmd.py:59]: I think this should be moved to a combiner config.yaml in due time where strategy parameters could be offset from default.

File: mnist-clients.yaml [hunk 2]
```diff
@@ -11,6 +11,10 @@ services:
       - FEDN_MINIO_ACCESS_KEY=${FEDN_MINIO_ACCESS_KEY}
       - FEDN_MINIO_SECRET_KEY=${FEDN_MINIO_SECRET_KEY}
     image: "scaleoutsystems/client:latest"
+    extra_hosts:
+      reducer: 130.239.81.136
+      combiner: 130.239.81.136
+      combiner2: 130.239.81.136
```
Comment (morganekmefjord) [inline mnist-clients.yaml:17]: Dont checkin!

File: mnist-clients.yaml [hunk 3]
```diff
@@ -31,6 +35,10 @@ services:
...
```
Comment (morganekmefjord) [inline mnist-clients.yaml:41]: Nono.

File: mnist-clients.yaml [hunk 4]
```diff
@@ -49,6 +57,10 @@ services:
...
```
Comment (morganekmefjord) [inline mnist-clients.yaml:63]: Njet

File: mnist-clients.yaml [hunk 5]
```diff
@@ -67,6 +79,10 @@ services:
...
     build:
```
Comment (morganekmefjord) [inline mnist-clients.yaml:86]: you get the point

**Review by morganekmefjord — CHANGES_REQUESTED** [review]

File: fedn/cli/run_cmd.py (hunk 1 above)
Comment (ahellander) [reply fedn/cli/run_cmd.py:59]: Agree. Lets create a ticket for it.

File: mnist-clients.yaml (hunk 2 above)
Comment (ahellander) [reply mnist-clients.yaml]: oops. Fixed.

**Pull request merged**